components/omega/doc/design/VertCoord.md

```diff
@@ -0,0 +1,294 @@
+The height of a layer ($\Delta z$) can be found using the mass thickness, density and known bottom elevation relative to the geoid (positive up).
+The vertical coordinate module will also serve as a container for information related to the variable number of vertical levels for a given ocean column due to variations in bottom elevation and ice shelf cavities.
+
+
+## 2 Requirements
+
+### 2.1 Requirement: Compute and store the total pressure at a given level
+
+The total pressure at the bottom of each level will be computed and stored within the vertical coordinate module.
+This involves a surface-down summation of the mass thickness variable times the gravitational acceleration, $g$, starting with the surface pressure.
+The pressure variable will be needed in the computation of the TEOS-10 equation of state and the pressure gradient.
+The variable will be registered with the `IOStreams` framework to enable it to be output during runtime.
+
+### 2.2 Requirement: Compute and store the $z$ location of a given level
+
+The $z$ location at the top of each level will be computed and stored within the vertical coordinate module.
+In the model, $z$ is defined as being positive up from the geoid.
+The $z$ location is calculated in a bottom-up summation of the mass thickness times the specific volume, starting with the known bottom elevation.
+The $z$ location of the interfaces is needed to compute the geopotential gradient and the sea level.
+The variable will be registered with the `IOStreams` framework to enable it to be output during runtime.
+
+### 2.3 Requirement: Compute and store the vertical level bounds for each horizontal cell
+
+All tendency terms require a variable vertical loop range to account for variation in bottom elevation and ice shelf cavities.
+These bounds will be computed in the vertical coordinate module and used for setting masking arrays that will be used to determine where calculations are valid in the vertical.
+
+### 2.4 Requirement: Compute and store the geopotential
+
+The geopotential will be computed and stored within the vertical coordinate module.
+Initially this will be calculated by summing together the $z$ location times $g$.
+In the future, tidal potential and self attraction and loading will also have contributions to the geopotential.
+
+### 2.5 Requirement: Compute and store the desired thickness used in the $p^\star$ vertical coordinate
+
+The $p^\star$ vertical coordinate specifies a desired layer thickness that stretches the initial thicknesses by the bottom pressure anomaly.
+This involves adding a vertically-weighted bottom pressure perturbation to the reference (initial) thickness.
+This desired thickness will be computed and stored within the vertical coordinate module.
+The vertical advection algorithm will use this desired thickness to compute the vertical transport.
+
+### Desired: General Arbitrary Lagrangian Eulerian (ALE) coordinate support
```

Comment:
Maybe add numbers to these too? The way it looks in the compiled version it kind of looks like these are a new section instead of just a part of Section 2.

Reply:
yes, thanks for pointing that out.